apps/next/src/components/TokenSelect/TokenSelect.tsx

```diff
@@ -34,16 +41,99 @@ function TokenSelectRaw({
   disabled,
 }: TokenSelectProps) {
   const { t } = useTranslation();
+  const { getNetwork } = useNetworkContext();
+  const [selectedChainId, setSelectedChainId] = useState<
+    number | 'avalanche' | null
+  >(null);
+
+  // Helper to check if a chain ID is any Avalanche network (C, X, or P)
+  const isAnyAvalancheNetwork = useCallback(
+    (chainId: number): boolean => {
+      const network = getNetwork(chainId);
+      return (
+        isAvalancheChainId(chainId) ||
+        isXchainNetwork(network) ||
+        isPchainNetwork(network)
```

Comment:
We have `isPchainNetworkId` and `isXchainNetworkId` functions, no need to `getNetwork` first.